dashboard/test/ui/step_definitions/steps.rb

```diff
@@ -538,20 +545,28 @@ def type_into_selector(input_text, selector)
   element_has_id(selector, id)
 end
 
+def jquery_element_exists(selector)
+  "return $(#{selector.dump}).length > 0"
+end
+
+def element_exists?(selector)
+  @browser.execute_script(jquery_element_exists(selector))
+end
+
+def element_visible?(selector)
+  @browser.execute_script(jquery_is_element_visible(selector))
+end
+
 Then /^element "([^"]*)" is (not )?visible$/ do |selector, negation|
-  visibility = @browser.execute_script("return $(#{selector.dump}).css('visibility')")
-  visible = @browser.execute_script("return $(#{selector.dump}).is(':visible')") && (visibility != 'hidden')
```

Comment:
Here we checked both conditions, but for some reason were making two round trips to the remote browser to do so.  This has been reduced to a single expression and a single request to the remote browser in the refactor.  Every little bit counts.